### PR TITLE
add signature algorithm argument to tpm2_sign to enable RSA-PSS signatures

### PR DESCRIPTION
**Pull request by zt-initech**
Closes: #1340

### CODE REVIEW

File: man/tpm2_sign.1.md [hunk 1]
```diff
@@ -39,6 +39,16 @@ data and validation shall indicate that hashed data did not start with
     Also, see section "Supported Hash Algorithms" for a list of supported hash
     algorithms.
 
+  * **-e**, **--signscheme**=_SIGNING\_SCHEME_:
```
Comment (williamcroberts) [inline man/tpm2_sign.1.md]: change the current `-s` to `-o` | `--out-sig` and use `-s` | `--sig-scheme` for this.

**Comment by williamcroberts** [comment]
Also make surer you update [analyze.py](https://github.com/tpm2-software/tpm2-tools/blob/master/scripts/utils/analyze.py) you can look for examples in PR https://github.com/tpm2-software/tpm2-tools/pull/1337
Travis Error:
```
group: signing:
	tools: ['tpm2_verifysignature', 'tpm2_sign']
	conflicts: set(['e', 'signscheme'])
The command "./.ci/travis.run" exited with 1.
```

**Comment by zt-initech** [comment]
@williamcroberts I've added test code to generate rsassa, rsapss and ecdsa signatures using the TPM and verify them using openssl, it also tests that the default scheme for rsa is rsassa.

File: tools/tpm2_sign.c [hunk 2]
```diff
@@ -175,6 +172,20 @@ static bool init(ESYS_CONTEXT *ectx) {
         }
     }
 
+    if (!ctx.flags.s) {
+        ctx.sig_scheme = TPM2_ALG_NULL;
+    }
```
Comment (williamcroberts) [inline tools/tpm2_sign.c]: Set this to the default in the context structure.

Comment (williamcroberts) [reply tools/tpm2_sign.c]: Done on https://github.com/tpm2-software/tpm2-tools/pull/1352

**Comment by williamcroberts** [comment]
merged on #1350

**Pull request closed without merging**